spring-cloud-aws-sqs/src/test/java/io/awspring/cloud/sqs/BaseSqsIntegrationTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sqs;
+
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.SQS;
+
+import com.amazonaws.auth.AWSCredentials;
+import java.io.IOException;
+import org.junit.jupiter.api.BeforeAll;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+
+@Testcontainers
+abstract class BaseSqsIntegrationTest {
+
+	protected static final String RECEIVES_MESSAGE_QUEUE_NAME = "receives.message.test.queue";
+	protected static final String DOES_NOT_ACK_ON_ERROR_QUEUE_NAME = "does.not.ack.test.queue";
+	protected static final String RESOLVES_PARAMETER_TYPES_QUEUE_NAME = "resolves.parameter.test.queue";
+	protected static final String RESOLVES_POJO_TYPES_QUEUE_NAME = "resolves.pojo.test.queue";
+	protected static final String RECEIVE_FROM_MANY_1_QUEUE_NAME = "receive.many.test.queue.1";
+	protected static final String RECEIVE_FROM_MANY_2_QUEUE_NAME = "receive.many.test.queue.2";
+	protected static final String ASYNC_RECEIVE_FROM_MANY_1_QUEUE_NAME = "async.receive.many.test.queue.1";
+	protected static final String ASYNC_RECEIVE_FROM_MANY_2_QUEUE_NAME = "async.receive.many.test.queue.2";
+
+	@Container
+	static LocalStackContainer localstack = new LocalStackContainer(
+			DockerImageName.parse("localstack/localstack:0.14.0")).withServices(SQS).withReuse(true);
```

Comment:
I don't think we want to use withReuse
```suggestion
			DockerImageName.parse("localstack/localstack:0.14.0")).withServices(SQS);
```

Reply:
Why not?

Reply:
It might not be related to this PR, but the question has been raised I want to share what I have in Spring Integration AWS for Localstack + Testcontainers: https://github.com/spring-projects/spring-integration-aws/blob/main/src/test/java/org/springframework/integration/aws/LocalstackContainerTest.java

Reply:
just replied here

https://github.com/awspring/spring-cloud-aws/pull/374#discussion_r956559082